core/src/apps/solana/layout.py

```diff
@@ -111,14 +113,19 @@ async def confirm_instruction(
 
             account_data: list[tuple[str, str]] = []
             if len(account_value) == 2:
```

Comment:
Unrelated to your changes, but please add a comment here like `# account included in transaction directly` and below at the `== 3` `if` a comment like `# lookup table address reference`. If see multiple instances of such checks in the code, we might want to introduce helpers like `isAccount` and `isLookupTableAccountReference` or something.